internal/provider/vaultradar/resource_radar_source_github_enterprise.go

```diff
@@ -0,0 +1,235 @@
+package vaultradar
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+
+	radar_service "github.com/hashicorp/hcp-sdk-go/clients/cloud-vault-radar/preview/2023-05-01/client/data_source_registration_service"
+
+	"github.com/hashicorp/terraform-plugin-framework-validators/stringvalidator"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/hashicorp/terraform-provider-hcp/internal/clients"
+	"github.com/hashicorp/terraform-provider-hcp/internal/provider/modifiers"
+)
+
+var (
+	_ resource.Resource              = &sourceGitHubEnterpriseResource{}
+	_ resource.ResourceWithConfigure = &sourceGitHubEnterpriseResource{}
+)
+
+func NewSourceGitHubEnterpriseResource() resource.Resource {
+	return &sourceGitHubEnterpriseResource{}
+}
+
+type sourceGitHubEnterpriseResource struct {
+	client *clients.Client
+}
+
+func (r *sourceGitHubEnterpriseResource) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_vault_radar_source_github_enterprise"
+}
+
+func (r *sourceGitHubEnterpriseResource) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		MarkdownDescription: "This Vault Radar resource manages HCP Cloud scans of GitHub Enterprise Server sources.",
```

Comment:
nit: Same here, managing scans is somewhat confusing. Consider changing to `This terraform resource manages GitHub Enterprise server datasource lifecycle in Vault Radar`?

Reply:
The other spot is generated from here: As noted I will update this to: 

> This terraform resource manages GitHub Enterprise Server data source lifecycle in Vault Radar.